src/BuildScriptGenerator/SdkStorageVersionProviderBase.cs

```diff
@@ -97,9 +98,25 @@ protected string GetDefaultVersion(string platformName, string sdkStorageBaseUrl
             this.logger.LogDebug("Getting the default version from url {defaultVersionUrl}.", defaultVersionUrl);
 
             // get default version
-            var defaultVersionContent = httpClient
-                .GetStringAsync($"{defaultVersionUrl}{this.commonOptions.OryxSdkStorageAccountAccessToken}")
-                .Result;
+            string defaultVersionContent;
+            try
+            {
+                defaultVersionContent = httpClient
+                    .GetStringAsync($"{defaultVersionUrl}{this.commonOptions.OryxSdkStorageAccountAccessToken}")
+                    .Result;
+            }
+            catch (AggregateException ae)
+            {
+                throw new AggregateException(
+                    $"Http request to retrieve the default version from '{defaultVersionUrl}' failed. " +
+                    $"{Constants.NetworkConfigurationHelpText}{Environment.NewLine}{ae}");
+            }
+
+            if (string.IsNullOrEmpty(defaultVersionContent))
+            {
+                throw new InvalidOperationException(
+                    $"Http request to retrieve the default version from '{defaultVersionUrl}' cannot return an empty result.");
+            }
```

Comment:
Yep makes sense. I _generally_ will try to follow the rule of 3, and so I thought this would be fine as is. Getting close to that point though. If people feel strongly, I'd be happy to spend some more time investing a bit more by extending httpclient.

Reply:
No need -- just an idea if we do end up reusing this snippet again in the future for any additional calls, but for now it's good as-is 👍